docs/source/create_an_experiment_(development).rst

```diff
@@ -0,0 +1,723 @@
+chain of calls is like this:
+
+::
+
+	transmit() -> _what() -> create_information() -> _contents().
+
+This might seem like a roundabout way to get the information, but it allows
+us to override any of the steps and return different information types or
+other modifications. Much of Dallinger is designed in this way, making it
+easy to create compatible, but perhaps completely different versions of its
+main constructs.
+
+The Experiment Code
+^^^^^^^^^^^^^^^^^^^
+
+Now that we are done setting up the experiment's infrastructure, we can
+write the code that will drive the actual experiment. Dallinger is very
+flexible, and you can design really complicated experiments for it. Some
+will require pretty heavy backend code, and probably a handful of
+dependencies. For this kind of advanced experiments, a lot of the code
+could be in Python.
+
+Dallinger also includes a Redis-based chat backend, which can be used to
+relay messages from experiment participants to the application and each
+other. All you have to do to enable this is to define a `channel` class
+variable with a string prefix for your experiment, and then you can use the
+experiment's `send` method to handle messages. Using this backend, you
+can easily create chat-enabled experiments, and even graphical UIs that
+can communicate user actions using channel messages.
+
+For this tutorial, however, we are keeping it simple, and thus will not
+require any other Python code for it. We already have a source for the texts
+defined, the network is set up, and recruitment is enabled, so all we need
+to get the Bartlett experiment going is a simple Javascript UI.
+
+The code that we will walk through will be saved in our ``experiment.js`` file:
+
+.. code-block:: javascript
+
+  var my_node_id
+
+  // Consent to the experiment.
+  $(document).ready(function() {
+
+    dallinger.preventExit = true;
+
+The ``experiment.js`` file will be executed on page load (see below for the
+template walk through), so we use the JQuery `$(document).ready` hook to
+run our code.
+
+The very first thing we do is setting `dallinger.preventExit` to True, which
+will prevent experiment participants from closing the window or reloading the
+page. This is to avoid the experiment being interrupted and the leaving the
+participant in an inconsistent state.
+
+Next, we define a few functions that will be called from the various
+experiment templates. This are functions that are more or less required for
+all experiments:
+
+.. code-block:: javascript
+
+    $("#print-consent").click(function() {
+      window.print();
+    });
+
+    $("#consent").click(function() {
+      store.set("recruiter", dallinger.getUrlParameter("recruiter"));
+      store.set("hit_id", dallinger.getUrlParameter("hit_id"));
+      store.set("worker_id", dallinger.getUrlParameter("worker_id"));
+      store.set("assignment_id", dallinger.getUrlParameter("assignment_id"));
+      store.set("mode", dallinger.getUrlParameter("mode"));
+
```

Comment:
We don't want to do this anymore. These values will be set earlier by Dallinger, and we don't want to overwrite them. See https://github.com/Dallinger/Dallinger/pull/1557